core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1611,6 +1668,95 @@ private void validateTableModelAndCreateTypeCast(
             throw SqlException.position(0).put("timestamp is not defined");
         }
     }
+    
+    private CompiledQuery sqlBackup(SqlExecutionContext executionContext) throws SqlException {
+		final CharSequence tok = SqlUtil.fetchNext(lexer);
+		if (null != tok) {
+			if (Chars.equalsLowerCaseAscii(tok, "table")) {
+				return sqlTableBackup(executionContext);
+			}
+		}
+		throw SqlException.position(lexer.getPosition()).put(" expected 'table'");
+      }
+    
+	private CompiledQuery sqlTableBackup(SqlExecutionContext executionContext) throws SqlException {
+		ObjList<CharSequence> tableNames = new ObjList<>();
+		while (true) {
+			CharSequence tok = SqlUtil.fetchNext(lexer);
+			if (null == tok) {
+				throw SqlException.position(lexer.getPosition()).put(" expected a table name");
+			}
+			CharSequence tableName = GenericLexer.unquote(tok);
+			int status = engine.getStatus(executionContext.getCairoSecurityContext(), path, tableName, 0, tableName.length());
+			if (status != TableUtils.TABLE_EXISTS) {
+				throw SqlException.position(lexer.getPosition()).put(" '").put(tableName).put("' is not  a valid table");
+			}
+			tableNames.add(tableName);
+
+			tok = SqlUtil.fetchNext(lexer);
+			if (null == tok || Chars.equals(tok, ';')) {
+				break;
+			}
+			if (!Chars.equals(tok, ',')) {
+				throw SqlException.position(lexer.getPosition()).put(" expected ','");
+			}
+		}
+		
+		for (int n=0; n<tableNames.size(); n++) {
+			backupTable(tableNames.get(n), executionContext);
+		}
+
+		return compiledQuery.ofRenameTable();
+	}
+    
+	public void backupTable(@NotNull CharSequence tableName, @NotNull SqlExecutionContext executionContext) {
+		LOG.info().$("Starting backup of ").$(tableName).$();
+		if (null == configuration.getBackupRoot()) {
+			throw CairoException.instance(0).put("Backup is disabled, no backup root directory is configured in the server configuration ['cairo.sql.backup.root' property]");
+		}
+
+		CairoSecurityContext securityContext = executionContext.getCairoSecurityContext();
+		try (TableReader reader = engine.getReader(securityContext, tableName)) {
+			cloneMetaData(tableName, configuration.getRoot(), configuration.getBackupRoot(), configuration.getMkDirMode(), (TableReaderMetadata) reader.getMetadata());
+			try (TableWriter backupWriter = engine.getBackupWriter(securityContext, tableName); RecordCursor cursor = reader.getCursor()) {
```

Comment:
reader.getCursor() is an oddball here. It must not be closed, otherwise it detaches from reader.

Obviously we need a test that shows backup can run multiple times